doc/source/whatsnew/v3.0.0.rst

```diff
@@ -663,9 +663,9 @@ Interval
 
 Indexing
 ^^^^^^^^
+- Bug in :class:`Index` printed names do not escape single quotes (:issue:`60190`)
```

Comment:
I do not think ``:class:`Index` printed names`` is clear, is there an objection to what I suggested in my previous comment? In any case, perhaps this is better.

    Bug in printing :attr:`Index.names` and :attr:`MultiIndex.levels` would not escape single quotes (:issue:`60190`)

Reply:
No there was not, in any case I will be directly including your current suggestion since it also include levels :) I'll make another PR.